core/server/services/settings/index.js

```diff
@@ -4,16 +4,78 @@
  */
 
 var SettingsModel = require('../../models/settings').Settings,
-    SettingsCache = require('./cache');
+    SettingsCache = require('./cache'),
+    SettingsLoader = require('./loader'),
+    _ = require('lodash'),
+    common = require('../../lib/common');
+    // SettingsFiles = require('./ensure-settings');
 
 module.exports = {
     init: function init() {
+        // TODO: the ensure settings fn up before the settings cache
+        // return SettingsFiles(this.knownSettings()).then(() => {
+
         // Update the defaults
         return SettingsModel.populateDefaults()
             .then(function (settingsCollection) {
                 // Initialise the cache with the result
                 // This will bind to events for further updates
                 SettingsCache.init(settingsCollection);
             });
+        // });
+    },
+
+    // TODO: is this the right place for it? We need a global place where
+    // we can add additional files that are supported.
+    // This can also be a config value for example.
+    knownSettings: function knownSettings() {
+        return ['routes'];
+    },
+
+    /**
+     * Getter for YAML settings objects.
+     * Example: `settingsSrvc.get('routes').then(...)`
+     * @param {String} setting type of setting. Needs to be supported.
+     *                         see `knownSettings`
+     * @returns {Promise<Object>} settingsFile
+     * @description Returns settings object as defined per YAML files in
+     * `/content/settings` directory.
+     */
+    get: function get(setting) {
+        const knownSettings = this.knownSettings();
+
+        if (!setting || _.indexOf(knownSettings, setting) < 0) {
+            return Promise.reject(new common.errors.IncorrectUsageError({
+                message: `Requested setting is not available: '${setting}'.`,
+                help: `Please use only the supported settings: ${knownSettings}.`
+            }));
+        }
+
+        return SettingsLoader()
```

Comment:
In theory, `settings.get('routes').then` will load all available settings files from disk, transform all files from yml to json and then returns the specific requested setting. Is that wanted?
I think we can leave it like that, but just wondered why it is like that.

Reply:
> In theory, settings.get('routes').then will load all available settings files from disk, transform all files from yml to json and then returns the specific requested setting. Is that wanted?

Yes. I am not happy with that either. I thought about this problem a lot, but couldn't come up with a solution, unless we access only and directly the file we want. e. g. we support `routes` in the `knownSettings` and only load this then. I think at one point, I had it coded like that but got insecure because of this sentence:

> Each file inside the content/settings folder needs to be read, parsed as YAML, and if it's valid, loaded as available settings for the services inside of Ghost.

Maybe I took it to much word by word here?

The only thing is: if we would change it to a logic that we only load and parse the requested settings, than we would need a different logic for `getAll()`. But you said already, that this is low priority for now.

Reply:
I think another reason that I changed the code to what it is now was that I wanted to make it more generic. Accessing the `content/settings` directory for only a particular file requires our logic, that this file is always there and I wanted to make it more independent... ¯\_(ツ)_/¯

Reply:
> The only thing is: if we would change it to a logic that we only load and parse the requested settings, than we would need a different logic for getAll()

In the loader: `getAll` can call `getOne` for each file?

Reply:
> In the loader: getAll can call getOne for each file?

👀 WOW. I didn't see that AT ALL.... 😂

Reply:
I will definitely change it to that, as I think it makes way more sense! Thanks!

Reply:
The code itself is changed, but I still need to update the tests and test it manually for myself.